crates/red_knot/src/lint.rs

```diff
@@ -357,4 +363,24 @@ mod tests {
             ]
         );
     }
+
+    /// Test that all snippets in testcorpus can be linted without panic
+    #[test]
+    #[allow(clippy::print_stdout)]
+    fn corpus_no_panic() {
+        let corpus = PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("resources/test/corpus");
+        let system_corpus =
+            SystemPathBuf::from_path_buf(corpus.clone()).expect("corpus path to be UTF8");
+        let mut db = setup_db_with_root(system_corpus.clone());
+        db.use_system(OsSystem::new(&system_corpus));
+
+        for path in fs::read_dir(&corpus).expect("corpus to be a directory") {
+            let path = path.expect("path to not be an error").path();
+            println!("checking {path:?}");
+            let path = SystemPathBuf::from_path_buf(path.clone()).expect("path to be UTF-8");
+            // this test is only asserting that we can run the lint without a panic
+            let file = system_path_to_file(&db, path).expect("file to exist");
+            lint_semantic(&db, file);
+        }
+    }
```

Comment:
I'm leaning towards making these integration tests (move the test to its own file in the `tests` directory instead of in `src`. 


https://doc.rust-lang.org/book/ch11-03-test-organization.html